src/main/java/StringAddCalculator.java

```diff
@@ -0,0 +1,56 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author jeongheekim
+ * @date 3/10/24
+ */
+public class StringAddCalculator {
+    public static int splitAndSum(String str) {
+        if (str == null || str.isEmpty()) {
+            return 0;
+        }
+        str = formattingStr(str);
+        return sum(split(str, extractSeperator(str)));
```

Comment:
str 이라는 변수가 4번이나 들어가는군요..!
메모리를 재활용하는 방식으로 구현하신것 같은데, 이런 경우 새롭게 변수를 재할당하는 것이 가독성이 좋을 수 있어요!

그리고 선언된 값을 중간에 값을 변경하거나 재활용하는 방식은 안티패턴으로 보기도 합니다!

ex)

```java
String a = "a";
if(any) {
  a = "b"
}
return a;
```

위와 같은 코드는

```java
if(any) {
  return "b"
}
return "a";
```

로 리팩토링될 수 있습니다.

splitAndSum 메서드를 리팩토링해보자면,

```suggestion
        String formattedStr = formattingStr(str);
        String seperator = extractSeperator(str);
        return sum(split(formattedStr, seperator));
```

이렇게 할 수 있을것 같은데, 이런 방식으로 구현해보시는건 어떠신가용??

Reply:
하나의 변수를 계속 재활용 하면 어디서 변경되었는지 추적하기 힘들겠군요!